client/src/components/WriteComment/index.tsx

```diff
@@ -24,16 +38,79 @@ interface ILocalState {
     postId: string;
     authorId: string;
     onChangeComments: Array<{ postId: string, comment: string }>;
+    page: number;
+    searchResults: object[];
+    loaded: boolean;
+    loading: boolean;
 }
 
 interface IState {
     feed: ILocalState;
     comments: IComments;
+    search: ISearchState;
+}
+
+interface IWriteComponentState {
+    isModalOpen: boolean;
+    searchValue: string;
 }
 
 export class WriteComment extends React.Component<IProps> {
+
+    public state: IWriteComponentState = {
+        isModalOpen: false,
+        searchValue: '',
+    };
+    public timer: any;
+
+    public componentWillUnmount = (): void => {
+        clearTimeout(this.timer);
+        this.props.clearSearchResults();
+        this.setState({searchValue: ''});
+        this.props.onChangeComment(this.props.postId, '');
+    }
+
     public onCommentChange = (postId: string, event: React.ChangeEvent<HTMLTextAreaElement>): void => {
+        clearTimeout(this.timer);
+        const FIRST_PAGE = 1;
         this.props.onChangeComment(postId, event.target.value);
+        const mentions: any = event.target.value.match(MENTIONS_REGEX);
+        if (mentions) {
+            const query: string = mentions.pop();
+            this.setState({searchValue: query});
+            if (this.state.searchValue !== query) {
+                this.timer = setTimeout(() => {
+                    this.timer = null;
+                    this.props.getSearchResults(query.slice(1), FIRST_PAGE);
+                    this.toggle(query);
+                }, 500);
+            } else {
+                this.toggle('');
+            }
+        } else {
+            this.toggle('');
+        }
+    }
```

Comment:
it is the same code? isn't it?

Reply:
It is very similar.

Reply:
maybe it is better to make it as general function and call it a few times where u want?